mcr.cpp

```diff
@@ -1,104 +1,53 @@
+	if (game[2][0] == game[2][1] && game[2][1] == game[2][2] && (game[2][0] == 'X' || game[2][0] == 'O')) win = true;
+	// column
+	if (game[0][0] == game[1][0] && game[1][0] == game[2][0] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true;
+	if (game[0][1] == game[1][1] && game[1][1] == game[2][1] && (game[0][1] == 'X' || game[0][1] == 'O')) win = true;
+	if (game[0][2] == game[1][2] && game[1][2] == game[2][2] && (game[0][2] == 'X' || game[0][2] == 'O')) win = true;
+	// diagonal
+	if (game[0][0] == game[1][1] && game[1][1] == game[2][2] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true;
+	if (game[0][2] == game[1][1] && game[1][1] == game[2][0] && (game[0][2] == 'X' || game[0][2] == 'O')) win = true;
+	return win;
 }
 
-bool is_right_position(int i, int j, char game[3][3]){
-    if (game[i][j] == ' ') {
-        return true;
-    }
-    return false;
-}
-// good !
-// nice ！
 int  main(){
-//    行、列索引
-    int i, j;
-    //    棋盘
-    //    初始化问题 √
-    char game[3][3] = {{' ',' ',' '}, {' ',' ',' '}, {' ',' ',' '}}; // Tic-tac-toe
-    //    重复代码 √
-    char player1 = 'X';
-    char player2 = 'O';
-    bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first.
-    std::cout << "X = Player 1" << std::endl << "O = Player 2" << std::endl;
-    int n;
-    for (n=0; n<9; n++){
-        print_chessboard(game);
-        turn = !turn;  // use the not-operator to change true to false or false to true.
-        if (turn == false)
-            std::cout << "Player 1: ";
-        else
-            std::cout << "Player 2: ";
-        //           输入检测
-        while(true){
-            std::cout << "Which cell to mark? i:[1..3], j:[1..3]: ";
-               std::cin >> i >> j;
-            // 输入校验 √
-            bool flag1 = is_right_input(i, j);
-            // 位置校验 √
-            bool flag2 = is_right_position(i, j, game);
-            if(flag1 && flag2){
-               break;
-            }
-        }
-        // 记录player
-        if (turn == false)
-           game[i][j] = player1;
-        else
-           game[i][j] = player2;
-        if (isWin(game)){
-            std::cout << "Win!" << std::endl;
-            break; // need to terminate the problem
-        }
-    }
-    
-//    判断和棋逻辑错误 √
-    if (n==9 && !isWin(game)) // all celles with i=0 to 2 have been inputted above but no winner yet
-       std::cout << "Tie!" << std::endl;
+	int i, j;
+	char game[3][3] = {' '}; // Tic-tac-toe
+	char player1 = 'X';
+	char player2 = 'O';
+	bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first.
+	std::cout << "X = Player 1" << std::endl << "O = Player 2" << std::endl;
+       for (int n=0; n<9; n++){
+			turn = !turn;  // use the not-operator to change true to false or false to true.
+		    if (turn == false)
+				std::cout << "Player 1: ";
+			else
+				std::cout << "Player 2: ";
```

Comment:
I think it is necessary to verify the input nums.

Reply:
ok i fix it